cogs/ca.py

```diff
@@ -523,8 +525,62 @@ async def sim(
-                return await ctx.send('`Error: Rule not found`')
+            except TypeError:  # rule not found
+                # first, attempt to load rule form wiki
+                try:
+                    async with self.session.get(f"https://conwaylife.com/w/api.php?action=parse&format=json&page=RULE:{rule}") as resp:
+                        b = await resp.json()
+                        contents = b["parse"]["text"]["*"]
+                        d = contents.replace("<p>", "\n").replace("</p>", "\n")
+                        tmp = "@RULE" + "@RULE".join(d.split("@RULE")[1:])
+                    rulefile = tmp
+                    segmented = mutils.segment(tmp)
+                    rulename = segmented["RULE"].split("\n")[0]
+
+                    extract_number = re.compile("^\d+")
+                    success = False
+
+                    # attempt to obtain n_states from ruletable
+                    for line in segmented["TABLE"].splitlines():
+                        if line.replace(" ", "").startswith("n_states:"):
+                            x = line.replace("n_states", "").replace(":", "").strip()  # lazy coding
+                            n_states = extract_number.match(x).group()
+                            success = True
+                            break
+
+                    # similar thing to ruletree
+                    for line in segmented["TREE"].splitlines():
+                        if line.replace(" ", "").startswith("num_states="):
+                            x = line.replace("num_states", "").replace("=", "").strip()  # lazy coding
+                            n_states = extract_number.match(x).group()
+                            success = True
+                            break
+
+                    if not success:
+                        raise ValueError("Error: n_states not found")
+
+                    n_states = int(n_states)
+
+                    # obtain colors
+                    colors_segment = segmented["COLORS"].splitlines()
+                    colors = {}
+                    for line in colors_segment:
+                        while "  " in line:
+                            line = line.replace("  ", " ")
+                        sections = line.split(" ")
+                        if len(sections) < 4:  # not enough values
+                            continue
+                        sections = sections[0:4]
+                        try:
+                            for idx in range(4):
+                                sections[idx] = int(sections[idx])
+                        except ValueError:
+                            continue
+                        colors[str(sections[0])] = (sections[1], sections[2], sections[3])
+                except KeyError:  # rule not found
+                    return await ctx.send('`Error: Rule not found`')
+                except ValueError:
+                    return await ctx.send("`Error: Invalid rule`")
```

Comment:
```suggestion
                except ValueError as e:
                    return await ctx.send(str(e))
```

This way, the original error text makes it to the Discord user.